Source/WebCore/style/AnchorPositionEvaluator.cpp

```diff
@@ -226,6 +234,20 @@ static LayoutSize offsetFromAncestorContainer(const RenderElement& descendantCon
     return offset;
 }
 
+static LayoutSize scrollOffsetFromAncestorContainer(const RenderElement& descendant, const RenderElement& ancestorContainer)
+{
+    ASSERT(descendant.isDescendantOf(&ancestorContainer));
+
+    auto offset = LayoutSize { };
+    for (auto* ancestor = descendant.container(); ancestor; ancestor = ancestor->container()) {
+        if (auto* box = dynamicDowncast<RenderBox>(ancestor))
+            offset -= toLayoutSize(box->scrollPosition());
```

Comment:
This is probably wrong with RTL scrollers.

Reply:
I don't think so, at least not for any obvious reasons. This is the same adjustment done in RenderObject::offsetFromContainer (we reverse it in offsetFromAncestorContainer() and re-apply it in scrollOffsetFromAncestorContainer). 

There are certainly RTL issues with the current code (we fails some existing tests) but those are probably better looked separately.